torch/utils/data/sampler.py

```diff
@@ -82,7 +83,13 @@ class WeightedRandomSampler(Sampler):
     """
 
     def __init__(self, weights, num_samples, replacement=True):
-        self.weights = torch.DoubleTensor(weights)
+        if not isinstance(num_samples, _int_classes) or isinstance(num_samples, bool):
+            raise ValueError("num_samples should be an integeral value, but "
+                             "got num_samples={}".format(num_samples))
+        if not isinstance(replacement, bool):
+            raise ValueError("replacement should be a boolean value, but got"
+                             "got replacement={}".format(replacement))
```

Comment:
Idk I feel like this is going a bit too far 😕 We can't have type checks at the beginning of every constructor, or we'll add a ton of overhead. The docs contain the expected types and we'll never each 100% coverage.

Having them in C++, where we do them faster and need them to prevent segfaults is one thing, but I don't think that enforcing strict type constraints is considered good practice in Python. I can understand exceptions when they can prevent common mistakes, but passing `'Hello world!'` as `batch_size` doesn't strike me as one 😄

Reply:
In this particular case the overhead doesn't matter, as it's only called very occasionally, right?

Reply:
I think it makes sense to add them in the places where overhead doesn't matter and which makes user experience better.

Reply:
If it makes user experience better then sure. That's why I mentioned that it makes perfect sense in places where it's a common error. I'm only saying that we shouldn't get overly defensive for no reason

Reply:
@apaszke I think it is reasonable to at least check `batch_size` because we don't even throw an error when its type is wrong, and that the error behavior is unexpected: treated as if `batch_size = inf`.

Reply:
Well, passing in an unexpected type to a Python function is basically an equivalent to UB known from C++, so I'd argue this is not unreasonable 😄 But whatever, let's merge this.